source/docs/software/advanced-control/feedforward.rst

```diff
@@ -0,0 +1,175 @@
+
+.. note:: The Java feedforward components will calculate outputs in units determined by the units of the user-provided feedforward gains.  Users *must* take care to keep units consistent, as WPILibJ does not have a type-safe unit system.
+
+The ``SimpleMotorFeedforward`` class calculates feedforwards for mechanisms that consist of permanent-magnet DC motors with no external loading other than friction and inertia, such as flywheels and robot drives.
+
+To create a ``SimpleMotorFeedforward``, simply construct it with the required gains:
+
+.. note:: The ``kA`` gain can be omitted, and if it is, will default to a value of zero.  For many mechanisms, especially those with little inertia, it is not necessary.
+
+.. tabs::
+
+  .. code-tab:: java
+
+    // Create a new SimpleMotorFeedforward with gains kS, kV, and kA
+    SimpleMotorFeedforward feedforward = new SimpleMotorFeedforward(kS, kV, kA);
+
+  .. code-tab:: c++
+
+    // Create a new SimpleMotorFeedforward with gains kS, kV, and kA
+    // Distance is measured in meters
+    frc::SimpleMotorFeedforward<units::meters> feedforward(kS, kV, kA);
+
+To calculate the feedforward, simply call the ``calculate()`` method with the desired motor velocity and acceleration:
+
+.. note:: The acceleration argument may be omitted from the ``calculate()`` call, and if it is, will default to a value of zero.  This should be done whenever there is not a clearly-defined acceleration setpoint.
+
+.. tabs::
+
+  .. code-tab:: java
+
+    // Calculates the feedforward for a velocity of 10 units/second and an acceleration of 20 units/second^2
+    // Units are determined by the units of the gains passed in at construction.
+    feedforward.calculate(10, 20);
+
+  .. code-tab:: c++
+
+    // Calculates the feedforward for a velocity of 10 meters/second and an acceleration of 20 meters/second^2
+    // Output is in volts
+    feedforward.Calculate(10_mps, 20_mps_sq);
+
+``ArmFeedforward``
+------------------
+
+.. note:: In C++, the ``ArmFeedforward`` class assumes distances are angular, not linear.  The passed-in gains *must* have units consistent with the angular, or a compile-time error will be thrown.  ``kS`` and ``kCos`` should have units of ``volts``, ``kV`` should have units of ``volts * seconds / radians``, and ``kA`` should have units of ``volts * seconds^2 / radians``.  For more information on C++ units, see :ref:`docs/software/basic-programming/cpp-units:The C++ Units Library`.
```

Comment:
Should be `consistent with the angular unit` in the second sentence of this paragraph.